lib/opencensus/trace/span_builder.rb

```diff
@@ -0,0 +1,129 @@
+# Copyright 2017 OpenCensus Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+module OpenCensus
+  module Trace
+    ##
+    # Span represents a single span within a request trace.
+    #
+    class SpanBuilder
+      ##
+      # The context that can build children of this span.
+      #
+      # @return [RequestTrace]
+      #
+      attr_reader :context
+
+      ##
+      # The numeric trace ID, as a big-endian unsigned integer.
+      #
+      # @return [Integer]
+      #
+      def trace_id
+        context.trace_id
+      end
+
+      ##
+      # The numeric span ID, as a big-endian unsigned integer.
+      #
+      # @return [Integer]
+      #
+      def span_id
+        context.span_id
+      end
+
+      ##
+      # The numeric span ID of the parent, as a big-endian unsigned integer.
+      # May be 0 if this is a root span.
+      #
+      # @return [Integer]
+      #
+      def parent_span_id
+        context.parent.span_id
```

Comment:
This will error if there's no parent

Reply:
There will always be a parent (the context that created this SpanBuilder is the parent).